src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/services/CourtServiceTest.kt

```diff
@@ -1047,6 +1047,74 @@ class CourtServiceTest {
       val bookings = service.getVideoLinkBookingsForPrisonAndDateAndCourt("WWI", date, null)
       assertThat(bookings).isEmpty()
     }
+
+    @Test
+    fun `appointments with missing end times are filtered out`() {
+      whenever(prisonApiService.getScheduledAppointmentsByAgencyAndDate(anyString(), any()))
+        .thenReturn(
+          listOf(
+            ScheduledAppointmentDto(
+              id = 1001,
+              agencyId = "WEI",
+              locationId = 1001,
+              appointmentTypeCode = "VLB",
+              startTime = LocalDateTime.of(2020, 1, 1, 9, 40),
+              endTime = LocalDateTime.of(2020, 1, 1, 10, 0)
+            ),
+            ScheduledAppointmentDto(
+              id = 1002,
+              agencyId = "WEI",
+              locationId = 1002,
+              appointmentTypeCode = "VLB",
+              startTime = LocalDateTime.of(2020, 1, 1, 10, 0),
+              endTime = LocalDateTime.of(2020, 1, 1, 10, 30)
+            ),
+            ScheduledAppointmentDto(
+              id = 1003,
+              agencyId = "WEI",
+              locationId = 1003,
+              appointmentTypeCode = "VLB",
+              startTime = LocalDateTime.of(2020, 1, 1, 10, 30),
+              endTime = null
+            )
+          )
+        )
+      whenever(videoLinkBookingRepository.findByMainAppointmentIds(any()))
+        .thenReturn(
+          listOf(
+            VideoLinkBooking(
+              id = 1000L,
+              pre = VideoLinkAppointment(
+                id = 10001,
+                bookingId = 1000,
+                appointmentId = 1001,
+                court = "Wimbledon",
+                hearingType = HearingType.PRE
+              ),
+              main = VideoLinkAppointment(
```

Comment:
What happens if the main appointment isn't valid? 
Remove the booking from the result list?

Reply:
fixed that